[exporter/splunkhec] Flaky test TestReceiveBatchedMetrics

### Component(s)

exporter/splunkhec

### Describe the issue you're reporting

Seen in https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4599359368/jobs/8124669438#step:9:5419

```
=== RUN   TestReceiveBatchedMetrics/1_metric_event_per_payload_(configured_max_content_length_is_same_as_event_size)
    client_test.go:941: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter/client_test.go:941
        	Error:      	"{\"time\":2.002,\"host\":\"unknown\",\"event\":\"metric\",\"fields\":{\"k/n1\":\"vn1\",\"k/r0\":\"vr0\",\"k/r1\":\"vr1\",\"metric_name:gauge_double_with_dims\":1234.5678,\"metric_type\":\"Gauge\",\"k0\":\"v0\",\"k1\":\"v1\",\"k/n0\":\"vn0\"}}" does not contain "\"time\":1.001"
        	Test:       	TestReceiveBatchedMetrics/1_metric_event_per_payload_(configured_max_content_length_is_same_as_event_size)
    client_test.go:941: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter/client_test.go:941
        	Error:      	"{\"time\":3.003,\"host\":\"unknown\",\"event\":\"metric\",\"fields\":{\"k/r1\":\"vr1\",\"metric_name:gauge_double_with_dims\":1234.5678,\"metric_type\":\"Gauge\",\"k0\":\"v0\",\"k1\":\"v1\",\"k/n0\":\"vn0\",\"k/n1\":\"vn1\",\"k/r0\":\"vr0\"}}" does not contain "\"time\":2.002"
        	Test:       	TestReceiveBatchedMetrics/1_metric_event_per_payload_(configured_max_content_length_is_same_as_event_size)
    client_test.go:941: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/splunkhecexporter/client_test.go:941
        	Error:      	"{\"host\":\"unknown\",\"event\":\"metric\",\"fields\":{\"metric_name:gauge_double_with_dims\":1234.5678,\"metric_type\":\"Gauge\",\"k0\":\"v0\",\"k1\":\"v1\",\"k/n0\":\"vn0\",\"k/n1\":\"vn1\",\"k/r0\":\"vr0\",\"k/r1\":\"vr1\"}}" does not contain "\"time\":3.003"
        	Test:       	TestReceiveBatchedMetrics/1_metric_event_per_payload_(configured_max_content_length_is_same_as_event_size)
=== RUN   TestReceiveBatchedMetrics/2_metric_events_per_payload_(configured_max_content_length_is_twice_event_size)
=== RUN   TestReceiveBatchedMetrics/1_compressed_batch_of_2037_bytes,_make_sure_the_event_size_is_more_than_minCompressionLen=1500_to_trigger_compression
=== RUN   TestReceiveBatchedMetrics/200_events,_make_sure_that_we_produce_more_than_one_compressed_batch
=== RUN   TestReceiveBatchedMetrics/200_events,_make_sure_that_we_produce_only_one_compressed_batch_when_MaxContentLengthMetrics_is_0
--- FAIL: TestReceiveBatchedMetrics (0.14s)
    --- PASS: TestReceiveBatchedMetrics/all_metrics_events_in_payload_when_max_content_length_unknown_(configured_max_content_length_0) (0.00s)
    --- FAIL: TestReceiveBatchedMetrics/1_metric_event_per_payload_(configured_max_content_length_is_same_as_event_size) (0.01s)
    --- PASS: TestReceiveBatchedMetrics/2_metric_events_per_payload_(configured_max_content_length_is_twice_event_size) (0.00s)
    --- PASS: TestReceiveBatchedMetrics/1_compressed_batch_of_2037_bytes,_make_sure_the_event_size_is_more_than_minCompressionLen=1500_to_trigger_compression (0.04s)
    --- PASS: TestReceiveBatchedMetrics/200_events,_make_sure_that_we_produce_more_than_one_compressed_batch (0.06s)
    --- PASS: TestReceiveBatchedMetrics/200_events,_make_sure_that_we_produce_only_one_compressed_batch_when_MaxContentLengthMetrics_is_0 (0.02s)
```

--- a/exporter/splunkhecexporter/client_test.go
+++ b/exporter/splunkhecexporter/client_test.go
@@ -937,9 +937,15 @@ func TestReceiveBatchedMetrics(t *testing.T) {
 				if test.want.compressed {
 					validateCompressedContains(t, test.want.batches[i], got[i].body)
 				} else {
+					found := false
+
 					for _, expected := range test.want.batches[i] {
-						assert.Contains(t, string(got[i].body), expected)
+						if strings.Contains(string(got[i].body), expected) {
+							found = true
+							break
+						}
 					}
+					assert.True(t, found, "%s did not match any expected batch", string(got[i].body))
 				}
 			}
 		})

